### PR TITLE
Remove Deprecated delegate_belongs_to

### PR DESCRIPTION
**Pull request by macanudo527**
#### What? Why?

`delegate_belongs_to` was [deprecated and removed from Spree Core ](https://github.com/spree/spree/pull/8275)in favor of native Rails `delegate`.

This will help make the code cleaner and easier to troubleshoot but also allows for easier detection of missing foreign keys. 

Came up in

- https://github.com/openfoodfoundation/openfoodnetwork/pull/11548

#### What should we test?

Test Gateways:
- Visit a store page
- Add items to cart
- Click on Cart
- Click Checkout
- Select Stripe or Paypal Express Payment Gateway
- Confirm Order

Test product import:
- New product
- New variant for existing product
- Update existing variant

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/models/product_import/entry_validator.rb [hunk 1]
```diff
@@ -67,10 +67,15 @@ def enterprise_field
...
       new_variant = Spree::Variant.new(
-        entry.assignable_attributes.except('id', 'product_id', 'on_hand', 'on_demand',
-                                           'variant_unit', 'variant_unit_name',
-                                           'variant_unit_scale', 'primary_taxon_id')
+        { product_id: }
+        .merge(
+          entry.assignable_attributes.except('id', 'product_id', 'on_hand', 'on_demand',
+                                             'variant_unit', 'variant_unit_name',
+                                             'variant_unit_scale', 'primary_taxon_id')
+        )
```
Comment (dacook) [inline app/models/product_import/entry_validator.rb]: I think another way to express this is like this with the "double-splat" operator. Also we might as well load the desired attributes in a separate step. Ok, I couldn't help but give it a try, it's cathartic. Hope you don't mind :)

Comment (dacook) [reply app/models/product_import/entry_validator.rb]: Sorry, I just noticed you're working on this at the same time, I'll step away and check later 😉

Comment (macanudo527) [reply app/models/product_import/entry_validator.rb]: No, should be ok. I just rebased it. Give it a shot.

File: app/models/spree/variant.rb [hunk 2]
```diff
@@ -221,6 +221,10 @@ def save_default_price
       default_price.save if default_price && (default_price.changed? || default_price.new_record?)
     end
 
+    def find_or_build_default_price
+      default_price || build_default_price
+    end
```
Comment (dacook) [inline app/models/spree/variant.rb:226]: I'm curious why this is  needed now, when it wasn't before.
Hmm.. i guess this is what DelegateBelongsTo was automatically doing in `delegator_for`..

**Review by dacook — APPROVED** [review]
Looks good to me, thanks for cleaning this up!

File: app/models/product_import/entry_validator.rb (hunk 1 above)
Comment (dacook) [reply app/models/product_import/entry_validator.rb]: Thanks, done, let me know if the added commit looks ok to you.

File: app/models/spree/variant.rb (hunk 2 above)
Comment (macanudo527) [reply app/models/spree/variant.rb:226]: Yeah, there was a build if not available built into `delegate_belongs_to`. [Here is a little more information](https://github.com/spree/spree/pull/6667/commits/7e1ab357738e74fde5ba153c1ad5603fe5dd79be) if you are interested.

File: app/models/product_import/entry_validator.rb (hunk 1 above)
Comment (macanudo527) [reply app/models/product_import/entry_validator.rb]: Looks like a big improvement. It looks much nicer.

File: app/models/spree/variant.rb (hunk 2 above)
Comment (dacook) [reply app/models/spree/variant.rb:226]: Ah ha, thank you.

**Review by mkllnk — APPROVED** [review]
Cool. To be honest, I'm not 100% sure about the impact but hopefully it will all be good.

**Comment by macanudo527** [comment]
> Cool. To be honest, I'm not 100% sure about the impact but hopefully it will all be good.

Yeah, this is mostly so I don't have to do the extra filtering in the spec for #11548 I'm not sure if that was clear or not. Hopefully it will speed something up though?

**Comment by mkllnk** [comment]
> > Cool. To be honest, I'm not 100% sure about the impact but hopefully it will all be good.
> 
> Yeah, this is mostly so I don't have to do the extra filtering in the spec for https://github.com/openfoodfoundation/openfoodnetwork/pull/11548 I'm not sure if that was clear or not. Hopefully it will speed something up though?

Oh, sorry, I didn't mean the positive impact of this PR. The clean-up is great. I meant that I'm not sure about the potential risk of anything breaking because of this change because I don't completely understand how delegate_belongs_to works. But that's also a good reason to get rid of it. :wink:

**Comment by drummer83** [comment]
Hi @macanudo527,
Thanks for further improving our code.
I have tested the following:
- Checking out with Stripe :heavy_check_mark: 
- Checking out with PayPal :heavy_check_mark: 
- Importing products
  - Creating new products :heavy_check_mark: 
  - Creating new variants :heavy_check_mark: 
  - Updating variants :heavy_check_mark: 

It's all looking good. Merging! :tada:

**Pull request merged**